In Prefix we are really self-providing, so don't look back!
Patch by Heiko Przybyl

--- gcc/gcc.c
+++ gcc/gcc.c
@@ -1525,8 +1525,8 @@ static const char *gcc_libexec_prefix;
 #endif
 
 static const char *const standard_exec_prefix = STANDARD_EXEC_PREFIX;
-static const char *const standard_exec_prefix_1 = "/usr/libexec/gcc/";
-static const char *const standard_exec_prefix_2 = "/usr/lib/gcc/";
+static const char *const standard_exec_prefix_1 = "@GENTOO_PORTAGE_EPREFIX@/usr/libexec/gcc/";
+static const char *const standard_exec_prefix_2 = "@GENTOO_PORTAGE_EPREFIX@/usr/lib/gcc/";
 static const char *md_exec_prefix = MD_EXEC_PREFIX;
 
 static const char *md_startfile_prefix = MD_STARTFILE_PREFIX;
--- gcc/config/darwin.h
+++ gcc/config/darwin.h
@@ -1286,7 +1286,7 @@ void add_framework_path (char *);
       }						\
   } while (0)
 
-#define SYSROOT_PRIORITY PREFIX_PRIORITY_FIRST
+#define SYSROOT_PRIORITY PREFIX_PRIORITY_LAST
 /* APPLE LOCAL end isysroot 5083137 */
 
 /* Every program on darwin links against libSystem which contains the pthread
